aquasec/resource_image.go

```diff
@@ -698,30 +702,24 @@ func resourceImageCreate(ctx context.Context, d *schema.ResourceData, m interfac
 		return diag.FromErr(err)
 	}
 
-	//d.SetId(getImageId(image))
-
-	err1 := resourceImageRead(ctx, d, m)
-	if err1 == nil {
-		d.SetId(getImageId(image))
-	} else {
-		return err1
-	}
+	d.SetId(getImageId(image))
+	return resourceImageRead(ctx, d, m)
 
-	return nil
-	//return resourceImageRead(ctx, d, m)
 }
 
 func resourceImageRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var err error
 	c := m.(*client.Client)
-	image := expandImage(d)
+	id := d.Id()
```

Comment:
duplicate set id, also in line: 705

Reply:
705 is the create function this is the read function?